System.IO.Streams/MemoryStream.cs

```diff
@@ -335,6 +353,11 @@ public override void SetLength(long value)
         {
             EnsureOpen();
 
+            if(!CanWrite)
```

Comment:
please use ctrl + k + d to have all the formatting of the code correct
Also, why not having an internal function like this instead of the if each time::

```csharp
private void EnsureWritable()
{
    if (!CanWrite)
    {
        throw new NotSupportedException();
    }
}
```

Reply:
I usually go by the rule of 3, if I have to put something more than three times I put it in a function, though I suppose 3 times is probably enough to, i'll get this changed.